### PR TITLE
cyme: init at 1.5.0

### PR DESCRIPTION
**Pull request by h7x4**
## Description of changes

Cyme is a modern lsusb replacement, with libusb as it's backend and some extra features.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/cy/cyme/package.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+rustPlatform.buildRustPackage rec {
+  pname = "cyme";
+  version = "1.5.0";
+
+  src = fetchFromGitHub {
+    owner = "tuna-f1sh";
+    repo = "cyme";
+    rev = "v${version}";
+    sha256 = "sha256-UXh97pHJ9wa/xSslHLB7WVDwLKJYvLPgmPX8RvKrsTI=";
```
Comment (eclairevoyant) [inline pkgs/by-name/cy/cyme/package.nix:19]:
```suggestion
    hash = "sha256-UXh97pHJ9wa/xSslHLB7WVDwLKJYvLPgmPX8RvKrsTI=";
```

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

File: pkgs/by-name/cy/cyme/package.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  buildInputs = [ libusb1 udev ];
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = with lib; {
+    homepage = "https://github.com/tuna-f1sh/cyme";
+    description = "A modern cross-platform lsusb";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ h7x4 ];
+    platforms = platforms.unix;
```
Comment (wegank) [inline pkgs/by-name/cy/cyme/package.nix:34]:
```suggestion
```

Comment (h7x4) [reply pkgs/by-name/cy/cyme/package.nix:34]: The default seems to include stiff like `wasm32-wasi` though, which doesn't really make sense for this application.

**Review by wegank — APPROVED** [review]

File: pkgs/by-name/cy/cyme/package.nix (hunk 2 above)
Comment (wegank) [reply pkgs/by-name/cy/cyme/package.nix:34]: I will check how to enable darwin builds, which ofborg says is not the case.

**Pull request merged**